client/blocks/site/index.jsx

```diff
@@ -145,3 +147,10 @@ export default React.createClass( {
 		);
 	}
 } );
+
+export default connect( ( state, { site } ) => {
+	if ( isNumber( site ) ) { // Accept site IDs
```

Comment:
Can do. The rationale for using `site` both for objects and IDs was to guard against unpredictable/unintuitive behavior when providing both a `site` and a `siteId` prop. But that's probably not really a biggie, so I'll go ahead with what you suggested.